packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,3233 @@
+  /// Material [MenuBar].
+  ///
+  /// If true, then a [PlatformMenuBar] will be substituted with the same
+  /// [menus], [child], and [enabled] but none of the visual attributes will
+  /// be passed along.
+  final bool _isPlatformMenu;
+
+  @override
+  State<MenuBar> createState() => _MenuBarState();
+
+  @override
+  List<DiagnosticsNode> debugDescribeChildren() {
+    return <DiagnosticsNode>[...menus.map<DiagnosticsNode>((MenuItem item) => item.toDiagnosticsNode())];
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(FlagProperty('enabled', value: enabled, ifFalse: 'DISABLED'));
+    properties.add(DiagnosticsProperty<MenuBarController>('controller', controller, defaultValue: null));
+    properties.add(
+        DiagnosticsProperty<MaterialStateProperty<Color?>>('backgroundColor', backgroundColor, defaultValue: null));
+    properties.add(DoubleProperty('height', height, defaultValue: null));
+    properties.add(DiagnosticsProperty<EdgeInsets?>('padding', padding, defaultValue: null));
+    properties.add(DiagnosticsProperty<MaterialStateProperty<double?>>('elevation', elevation, defaultValue: null));
+  }
+}
+
+class _MenuBarState extends State<MenuBar> {
+  late FocusNode focusNode;
+  late Map<MenuSerializableShortcut, Intent> shortcuts;
+  _MenuBarController? _controller;
+  _MenuBarController get controller {
+    // Make our own controller if the user didn't provide one.
+    if (widget.controller != null) {
+      _controller?.dispose();
+      _controller = null;
+      return widget.controller! as _MenuBarController;
+    }
+    return _controller ??= _MenuBarController();
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    focusNode = FocusNode(debugLabel: 'MenuBar');
+    _updateController();
+  }
+
+  @override
+  void dispose() {
+    focusNode.dispose();
+    controller.disconnect();
+    // Only dispose of the controller if we created it in this state object.
+    _controller?.dispose();
+    super.dispose();
+  }
+
+  void _updateController() {
+    shortcuts = _getShortcuts();
+    controller.update(context: context, topLevel: widget.menus, enabled: widget.enabled);
+  }
+
+  @override
+  void didUpdateWidget(MenuBar oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.controller != oldWidget.controller) {
+      (oldWidget.controller as _MenuBarController?)?.disconnect();
+    }
+    if (widget.controller != oldWidget.controller || widget.menus != oldWidget.menus) {
+      _updateController();
+    }
+    controller.enabled = widget.enabled;
+  }
+
+  void _doSelect(Intent onSelected) {
+    if (controller.enabled) {
+      Actions.maybeInvoke(FocusManager.instance.primaryFocus!.context!, onSelected);
+    }
+    controller.closeAll();
+  }
+
+  /// These are only used in debug mode to make sure there aren't any duplicate
```

Comment:
It's more subtle than that.  The other comment (from shortcuts.dart, on `ShortcutRegistry.addAll`) says:

```dart
  /// If two equivalent, but different, [ShortcutActivator]s are added, all of
  /// them will be executed when triggered. For example, if both
  /// `SingleActivator(LogicalKeyboardKey.keyA)` and `CharacterActivator('a')`
  /// are added, then both will be executed when an "a" key is pressed.".
```

So if you have a `SingleActivator(LogicalKeyboardKey.keyA)` and `CharacterActivator('a')`, they aren't identical, and so they are allowed, but they both will trigger when you press "a".

Because of the way registering shortcuts works, it would be hard to register multiple intents for a single shortcut, and seems like most of the time that would be an error anyhow. I would ban the example above too if I could (just for sanity's sake), but I don't think I have a way to compare the activators to make sure they don't fire on the same conditions, especially if someone rolls their own activator subclass.